Userland/Libraries/LibJS/Bytecode/ASTCodegen.cpp

```diff
@@ -540,11 +540,11 @@ Bytecode::CodeGenerationErrorOr<void> SuperCall::generate_bytecode(Bytecode::Gen
             TRY(arg.value->generate_bytecode(generator));
             auto arg_reg = generator.allocate_register();
             generator.emit<Bytecode::Op::Store>(arg_reg);
-            argument_registers.unchecked_append(arg_reg);
+            argument_registers.unchecked_append({ arg_reg, arg.is_spread });
```

Comment:
I strongly suggest generating a loop and calling via a runtime-constructed array instead; your current approach will not behave correctly when sequencing can be observed:
```js
function bar(...xs) { return xs; }
let a = [];
bar(
    a.push(0),
    a.push(1),
    ...a,
    a.push(2)
); // should be [1, 2, 0, 1, 3]; you'll observe [1, 2, 0, 1, 2, 3]
```
your code would not generate the correct sequence point for that yield.